databuilder/databuilder/publisher/neo4j_csv_unwind_publisher.py

```diff
@@ -0,0 +1,501 @@
+        :return:
+        """
+        LOGGER.info('Creating indices. (Existing indices will be ignored)')
+
+        with open(node_file, 'r', encoding='utf8') as node_csv:
+            for node_record in pandas.read_csv(node_csv,
+                                               na_filter=False).to_dict(orient='records'):
+                label = node_record[NODE_LABEL_KEY]
+                if label not in self.labels:
+                    self._try_create_index(label)
+                    self.labels.add(label)
+
+        LOGGER.info('Indices have been created.')
+
+    def _write_transactions(self,
+                            records: List[dict],
+                            stmt: str,
+                            confirm_rel_created: bool = False) -> None:
+        params_list = []
+        start_idx = 0
+        while start_idx < len(records):
+            stop_idx = min(start_idx + self._transaction_size, len(records))
+
+            for i in range(start_idx, stop_idx):
+                params_list.append(self._create_props_param(records[i]))
+
+            with self._driver.session(database=self._db_name) as session:
+                session.write_transaction(self._execute_statement, stmt, {'batch': params_list},
+                                          expect_result=confirm_rel_created)
+
+            params_list.clear()
+            start_idx = start_idx + self._transaction_size
+
+    def _publish_node(self, node_file: str) -> None:
+        """
+        Iterate over the csv records of a file, each csv record transform to Merge statement
+        and will be executed.
+        All nodes should have a unique key, and this method will try to create unique index on
+        the LABEL when it sees first time within a job scope.
+        Example of Cypher query executed by this method:
+        MERGE (col_test_id1:Column {key: 'presto://gold.test_schema1/test_table1/test_id1'})
+        ON CREATE SET col_test_id1.name = 'test_id1',
+                      col_test_id1.order_pos = 2,
+                      col_test_id1.type = 'bigint'
+        ON MATCH SET col_test_id1.name = 'test_id1',
+                     col_test_id1.order_pos = 2,
+                     col_test_id1.type = 'bigint'
+
+        :param node_file:
+        :return:
+        """
+
+        with open(node_file, 'r', encoding='utf8') as node_csv:
+            all_node_records = pandas.read_csv(node_csv, na_filter=False).to_dict(orient="records")
+            if len(all_node_records) > 0:
+                stmt = self.create_node_merge_statement(node_record=all_node_records[0])
+
+            self._write_transactions(all_node_records, stmt)
+
+    def is_create_only_node(self, node_record: dict) -> bool:
+        """
+        Check if node can be updated
+        :param node_record:
+        :return:
+        """
+        if self.create_only_nodes:
+            return node_record[NODE_LABEL_KEY] in self.create_only_nodes
+        else:
+            return False
```

Comment:
return `self.create_only_nodes.get(node_record[NODE_LABEL_KEY], False)` can replace this if self.create_only_nodes defaults to empty dict. Also, if this is only a one liner, do we even need a separate function.